internal/provider/framework_provider.go

```diff
@@ -0,0 +1,210 @@
+					getCloudBuilderConflictsValidator(),
+					stringvalidator.AlsoRequires(
+						path.Expressions{
+							path.MatchRoot("cloud_builder_username"),
+							path.MatchRoot("cloud_builder_host"),
+						}...),
+				},
+			},
+			"cloud_builder_host": schema.StringAttribute{
+				Optional:    true,
+				Description: "Fully qualified domain name or IP address of the CloudBuilder",
+				Validators: []validator.String{
+					getCloudBuilderConflictsValidator(),
+					stringvalidator.AlsoRequires(
+						path.Expressions{
+							path.MatchRoot("cloud_builder_username"),
+							path.MatchRoot("cloud_builder_password"),
+						}...),
+				},
+			},
+			"allow_unverified_tls": schema.BoolAttribute{
+				Optional:    true,
+				Description: "If set, VMware VCF client will permit unverifiable TLS certificates.",
+			},
+		},
+	}
+}
+
+func (p *FrameworkProvider) Metadata(ctx context.Context, req provider.MetadataRequest, res *provider.MetadataResponse) {
+	res.TypeName = "vcf"
+}
+
+func (p *FrameworkProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		func() resource.Resource {
+			return &ResourceNetworkPool{}
+		},
+	}
+}
+
+func (p *FrameworkProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{}
+}
+
+func (p *FrameworkProvider) Configure(ctx context.Context, req provider.ConfigureRequest, res *provider.ConfigureResponse) {
+	var data FrameworkProviderModel
+
+	res.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	sddcManagerUsername := getAttributeValue(data.SddcManagerUsername.ValueString(), constants.VcfTestUsername).(string)
+
+	if sddcManagerUsername != "" {
+		// Connect to SDDC Manager
+		client := api_client.NewSddcManagerClient(
+			sddcManagerUsername,
+			getAttributeValue(data.SddcManagerPassword.ValueString(), constants.VcfTestPassword).(string),
+			getAttributeValue(data.SddcManagerHost.ValueString(), constants.VcfTestUrl).(string),
+			getAttributeValue(data.AllowUnverifiedTls.ValueBool(), constants.VcfTestAllowUnverifiedTls).(bool),
+		)
+
+		if err := client.Connect(); err != nil {
+			res.Diagnostics.Append(diag.NewErrorDiagnostic("Failed to connect to the SDDC Manager", err.Error()))
+		}
+
+		p.SddcManagerClient = client
+		res.ResourceData = client
+	} else {
+		// Connect to Cloud Builder
+		client := api_client.NewCloudBuilderClient(
+			getAttributeValue(data.CloudBuilderUsername.ValueString(), constants.CloudBuilderTestUsername).(string),
+			getAttributeValue(data.CloudBuilderPassword.ValueString(), constants.CloudBuilderTestPassword).(string),
+			getAttributeValue(data.CloudBuilderHost.ValueString(), constants.CloudBuilderTestUrl).(string),
+			getAttributeValue(data.AllowUnverifiedTls.ValueBool(), constants.VcfTestAllowUnverifiedTls).(bool),
+		)
+
+		p.CloudBuilderClient = client
+		res.ResourceData = client
+	}
+}
+
+func getAttributeValue[T string | bool](data T, envVar string) interface{} {
```

Comment:
I have the feeling you're going to use a lot of methods like these in this new version of the provider. Maybe you should extract them into a separate utils file?

Reply:
Data access is actually quite decent with the framework. This method was necessary because I wanted to retain the provider's capability to read its settings from environment variables (the standard library can only read them as strings).

Do you mind if we wait until we migrate at least one more resource? It may turn out that we only need this in this one place.